be/src/exprs/array_functions.tpp

```diff
@@ -130,28 +131,42 @@ public:
 
     static ColumnPtr process(FunctionContext* ctx, const Columns& columns) {
         RETURN_IF_COLUMNS_ONLY_NULL(columns);
-        if constexpr (lt_is_arithmetic<LT> || lt_is_decimalv2<LT>) {
-            return _array_difference(columns);
+        if constexpr (lt_is_float<LT>) {
```

Comment:
why write so many branch? _array_difference is already a template,  just use return _array_difference<LT>(ctx, columns);

Reply:
emm, result type is different with input type, the template parameter of `_array_difference` is the result type, but the template parameter of ArrayDifference is the input type